server/services/broadlink/lib/learn/broadlink.learn.js

```diff
@@ -0,0 +1,48 @@
+const { EVENTS, WEBSOCKET_MESSAGE_TYPES } = require('../../../../utils/constants');
+const logger = require('../../../../utils/logger');
+const { ACTIONS } = require('../utils/broadlink.constants');
+/**
+ * @description Entering peripheral in learn mode.
+ * @param {string} peripheralIdentifier - The peripheral address.
+ * @example
+ * await gladys.broadlink.learn('770f78b9401c');
+ */
+async function learn(peripheralIdentifier) {
+  logger.debug(`Broalink entering learn mode with ${peripheralIdentifier}`);
+  let peripheral;
+
+  try {
+    peripheral = await this.getDevice(peripheralIdentifier);
+  } catch (e) {
+    this.gladys.event.emit(EVENTS.WEBSOCKET.SEND_ALL, {
+      type: WEBSOCKET_MESSAGE_TYPES.BROADLINK.LEARN_MODE,
+      payload: {
+        action: ACTIONS.LEARN.NO_PERIPHERAL,
+      },
+    });
+  }
+
+  if (peripheral) {
+    try {
+      await peripheral.enterLearning();
+
+      // async check data
+      logger.debug(`Broadlink entering ${peripheralIdentifier} in learn mode...`);
+      this.checkData(peripheral, peripheralIdentifier);
```

Comment:
await is not required here, because this only starts the process, and response is handled as a callback.
So we start the process, it runs each 0.8s the check on device, once done, we send a WebSocket event to alert front side.